src/session/chat/history.rs

```diff
@@ -327,21 +328,39 @@ impl History {
 
     fn remove(&self, message_id: i64) {
         let self_ = imp::History::from_instance(self);
-        let index = self_.list.borrow().iter().position(|m| {
-            if let ItemType::Message(message) = m.type_() {
-                if message.id() == message_id {
-                    return true;
-                }
-            }
 
-            false
-        });
+        if let Some(message) = self_.message_map.borrow_mut().remove(&message_id) {
+            let message_date_time = glib::DateTime::from_unix_utc(message.date() as i64).unwrap();
```

Comment:
Just noticed this as well: this line can be moved in the DayDivider match block.